### ISSUE TITLE
RFE: management command to revoke an authentication token

### ISSUE THREAD

**Issue by gcoxmoz** [OpenVPN/openvpn#346]
tl;dr:
There is not currently a way for a server admin to revoke a token given out by `--auth-gen-token`.  I believe there should be.  Priority: low.

User Story:
I have a CentOS7 2.6.3 server where I have `--auth-gen-token 0`.  I have a hook in auth-user-pass-verify to do our OTP work.  Once verified, that token lives forever.

A main use case of these tokens is to avoid human-entering-OTP annoyance upon renegotiation, and it's not unreasonable for the `lifetime` parameter on `--auth-gen-token` to be "a fairly long time" (that is, on the order of 'days / weeks / forever').  While in hindsight it's pretty easy to say "'the token lives forever', the server did what you asked for, you have nobody to blame but yourself," you'd be pretty much right, but that misses the point: even if I'd done an almost-unreasonably-short "1 day" it's still a long time that someone is allowed in, if you are wanting them gone.

Back when our server was on 2.5.x, we didn't use `--auth-gen-token` (due to 2.4.4-era token issues) but rather `--reneg-sec 0`.  With that, every client that connected was going through an auth-user-pass-verify plugin.  If a user was exited, we could `client-kill` them through management, and any attempt to reconnect would bring them back to a place where they would not get in.  Great.
We switched to `--auth-gen-token 0` (and deleted/returned `--reneg-sec` to default) when we got rid of all 2.4 clients and upgraded the server to 2.6.  Recently we had a terminated user whose client was reconnecting automatically, and they were being allowed in because of their still-valid token.  The crux of the problem was that our client-connect script was not doing any authorization work, as that had all been deferred up to auth-user-pass-verify, and thus nothing was stopping the connection.

In the management area there is `client-kill`, which means we accept the idea of "I allowed you on before, but now I want your /session/ to end."  This RFE is the notion that, "I allowed you on before, but now I want your /credentials/ to end."  This doesn't come up often because, for most server admins, the former covers the latter, but there's an edge case here when using hooks.  Whether token revocation is part of client-kill or a new/similar command is debatable, and I don't have an opinion there.  I think token revocation should be admin-available in some manner, whatever it is.

There are other ways we could've solved this, in hindsight.  Alternatives considered (some before the fact, some after):
* Restarting the server: highly disruptive to kick everyone off.
* Revoking the user's certificates: Only works if you have a CA that can do a CRL and have the CRL plumbed in (which we didn't, for a few reasons of local-CA-complexity).  Even if trivially implementable, we were into needing a disruptive server restart to pick the `--crl` directive.
* Firewall the user's remote IP: effective in starting the 2*`reneg-sec` clock, but only if they don't change the IP, and requires sysadmin-solving externally to openvpn.
* Remove their CCD config: not in play in this situation.
* Use `external-auth` on `--auth-gen-token`: Probably the most correct, did not realize this option / consider in time.
* Put authorization code in `--client-connect`: what we chose since it was 'copying known code' under time pressure.

**Comment by selvanair** [comment]
`client-kill CID RESTART`issued from the server's management interface should do what you are asking for. Replace `CID` by the client-id. The client will restart and go through a new authentication cycle.

**Comment by gcoxmoz** [comment]
`kill CN` was called many times, and the user continued to reconnect.  I -think- I did `client-kill CID` once but I couldn't swear to it.  If kill vs client-kill behave differently, that's surprising and unclear to me.  If kill and client-kill are "the same idea but different client identifiers" then they exhibit the issue of passing the token again.
